### ISSUE TITLE
Mixing up multi threading with async io 

### ISSUE THREAD

**Issue by TwistingTwists** [pydelhi/talks#315]
### Title

Multithreading in python and async io

### Describe your Talk

When working with async io, what happens if you have sync function that needs to be run in the async event loop? 

### Pre-requisites & reading material

Basics of async io and os threads.  

### Time required for the talk

20 min

### Link to slides/demos

https://gist.github.com/TwistingTwists/b4d2ce7b4cab00b3f94f3829893777dc

Initial running slides. Polished version to come.

### About you

Software developer with 5+ years of experience. 

### Availability

21/12/2024

**Comment by Animesh-Ghosh** [comment]
Hey @TwistingTwists, thanks for the talk proposal. Please add a link to your slides in the original comment.

Thanks!

**Comment by TwistingTwists** [comment]
https://gist.github.com/TwistingTwists/b4d2ce7b4cab00b3f94f3829893777dc

running slides.

**Comment by TwistingTwists** [comment]
> Hey @TwistingTwists, thanks for the talk proposal. Please add a link to your slides in the original comment.
> 
> Thanks!

done ;)

**Comment by Animesh-Ghosh** [comment]
The gist looks good enough for a lightning talk IMO @agriyakhetarpal. If possible, you may talk about the overheads for both the strategies (ex. ProcessPoolExecutor might have to write and read back from a Process Control Block) for different scenarios @TwistingTwists.

TBF I don't see any mention of `asyncio` but I am assuming you'll give a small demo around using these two `concurrent.futures` modules with it?

I'd personally like to know more about this - and would appreciate it if you could update the gist and maybe upgrade to proper slides (Google Slides, Slides.com, whatever works for you).

**Comment by agriyakhetarpal** [comment]
I agree with @Animesh-Ghosh – the content should be alright for our audience, but going the extra mile with a small demo would be great. I'd love to hear such a talk as well.

I think we can accept this as a lightning talk for today (apologies for the short notice!), and if you wish to, we can have a longer, thirty-minute session in January.

**Comment by agriyakhetarpal** [comment]
Also, yes – if you can, please bring some slides!

**Comment by agriyakhetarpal** [comment]
Sorry to hear you weren't able to give this talk at this month's meetup, @TwistingTwists. Would you like a tentative slot for next month? We would be in a position to share the details next month, so there's no hurry on a response.

**Comment by TwistingTwists** [comment]
> Sorry to hear you weren't able to give this talk at this month's meetup, [@TwistingTwists](https://github.com/TwistingTwists). Would you like a tentative slot for next month? We would be in a position to share the details next month, so there's no hurry on a response.

Next month is possible. I'll prepare some slides / write up / blogpost.

**Comment by agriyakhetarpal** [comment]
Hi @TwistingTwists! How about 29/03/2025 – would you be available this month?

**Comment by agriyakhetarpal** [comment]
Hi again, @TwistingTwists! Would you be interested in delivering your talk in the May edition of the meetup? While the venue and logistics are undecided at the moment, it's slated to be on the 17th of May, and it could be in collaboration with [PyData Delhi](https://github.com/pydatadelhi/).

Please feel free to add a link to your slides/blogs/etc. whenever it's convenient for you to do so, and let us know your availability. Thank you!
